app/src/main/java/org/simple/clinic/wip/PrimarySolidButtonDrawable.kt

```diff
@@ -0,0 +1,101 @@
+package org.simple.clinic.wip
+
+import android.content.res.Resources
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.Rect
+import android.graphics.RectF
+import android.graphics.drawable.ColorDrawable
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.GradientDrawable
+import android.graphics.drawable.StateListDrawable
+import android.os.Build
+import android.support.annotation.ColorInt
+import android.util.DisplayMetrics
+import timber.log.Timber
+
+/**
+ * WIP.
+ *
+ * This drawable supports reading drawables from Xml selectors, making it compatible
+ * with the standard way of applying different Drawable states. The current implementation of
+ * [PrimarySolidButton] involves manually setting the enabled and disabled colors as
+ * direct attributes on the button.
+ */
+class PrimarySolidButtonDrawable(delegate: StateListDrawable) : StateListDrawable() {
```

Comment:
FYI: this is a work in progress thing and can be skipped for review. I just want to check it in for future usage.